transport/peers/peer.go

```diff
@@ -0,0 +1,82 @@
+package peers
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new HostPortPeerIdentifier
+func NewPeerIdentifier(hostport string) *HostPortPeerIdentifier {
+	return &HostPortPeerIdentifier{
+		hostport: hostport,
+	}
+}
+
+// HostPortPeerIdentifier uniquely references a host:port combination using a common interface
+type HostPortPeerIdentifier struct {
+	hostport string
+}
+
+// Identifier generates a (should be) unique identifier for this PeerIdentifier (to use in maps, etc)
+func (p *HostPortPeerIdentifier) Identifier() string {
+	return p.hostport
+}
+
+// NewPeer creates a new HostPortPeer from a PeerIdentifier, Agent, and Subscriber
+func NewPeer(pid transport.PeerIdentifier, agent transport.PeerAgent, subscriber transport.PeerSubscriber) *HostPortPeer {
+	hppid := pid.(*HostPortPeerIdentifier)
+	return &HostPortPeer{
+		HostPortPeerIdentifier: HostPortPeerIdentifier{
+			hostport: hppid.hostport,
```

Comment:
Actually, if `hppid` is a `*HostPortPeerIdentifier`, you can do
`HostPortPeerIdentifier: *hppid`.